# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/gl/GLBlitHelperD3D.cpp
# Commit: edfb73b96d5a
# Full Hash: edfb73b96d5ad946bbdd9da2e98d07b1f4565e1f
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2019-06-15 21:44:24
# Regressor Bug: 1559524
# File Overlap Count: 2
# Description:
#   Bug 1559524 - Reduce calls to GLLibraryEGL::Get(). r=lsalzman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D35103
# ==============================================================================

diff -r 0378898329b2 -r edfb73b96d5a gfx/gl/GLBlitHelperD3D.cpp
--- a/gfx/gl/GLBlitHelperD3D.cpp	Sat Jun 15 05:14:53 2019 +0000
+++ b/gfx/gl/GLBlitHelperD3D.cpp	Sat Jun 15 05:21:12 2019 +0000
@@ -20,9 +20,9 @@
 namespace mozilla {
 namespace gl {
 
-static EGLStreamKHR StreamFromD3DTexture(ID3D11Texture2D* const texD3D,
+static EGLStreamKHR StreamFromD3DTexture(GLLibraryEGL* const egl,
+                                         ID3D11Texture2D* const texD3D,
                                          const EGLAttrib* const postAttribs) {
-  auto* egl = gl::GLLibraryEGL::Get();
   if (!egl->IsExtensionSupported(
           GLLibraryEGL::NV_stream_consumer_gltexture_yuv) ||
       !egl->IsExtensionSupported(
@@ -85,7 +85,8 @@
     MOZ_RELEASE_ASSERT(numPlanes >= 1 && numPlanes <= 3);
 
     const auto& gl = mParent.mGL;
-    auto* egl = gl::GLLibraryEGL::Get();
+    const auto& gle = GLContextEGL::Cast(gl);
+    const auto& egl = gle->mEgl;
     const auto& display = egl->Display();
 
     gl->fGenTextures(numPlanes, mTempTexs);
@@ -97,7 +98,7 @@
       if (postAttribsList) {
         postAttribs = postAttribsList[i];
       }
-      mStreams[i] = StreamFromD3DTexture(texD3DList[i], postAttribs);
+      mStreams[i] = StreamFromD3DTexture(egl, texD3DList[i], postAttribs);
       mSuccess &= bool(mStreams[i]);
     }
 
@@ -121,7 +122,8 @@
 
   ~BindAnglePlanes() {
     const auto& gl = mParent.mGL;
-    auto* egl = gl::GLLibraryEGL::Get();
+    const auto& gle = GLContextEGL::Cast(gl);
+    const auto& egl = gle->mEgl;
     const auto& display = egl->Display();
 
     if (mSuccess) {
@@ -150,7 +152,8 @@
 
   if (!mGL->IsANGLE()) return nullptr;
 
-  auto* egl = gl::GLLibraryEGL::Get();
+  const auto& gle = GLContextEGL::Cast(mGL);
+  const auto& egl = gle->mEgl;
   EGLDeviceEXT deviceEGL = 0;
   MOZ_ALWAYS_TRUE(egl->fQueryDisplayAttribEXT(
       egl->Display(), LOCAL_EGL_DEVICE_EXT, (EGLAttrib*)&deviceEGL));